driver-core/src/test/unit/com/mongodb/internal/connection/ScramShaAuthenticatorSpecification.groovy

```diff
@@ -387,6 +387,24 @@ class ScramShaAuthenticatorSpecification extends Specification {
         async << [true, false]
     }
 
+    def 'should skip empty exchange on server authentication SHA-256'() {
```

Comment:
Re-organize the tests so that the test for the old style and the new style of conversation are next to each other in the test file.  The test should be named something like

`should complete SCRAM conversation with no empty exchange`
`should complete SCRAM conversation with empty exchange`

Reply:
I have updated each test that should complete successfully to call a method that will run with and without an empty exchange.